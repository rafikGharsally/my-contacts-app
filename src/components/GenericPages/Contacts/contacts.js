import React, { Component } from 'react';

class contacts extends Component {

  render () {

  const { contacts } = this.props;

  return (
    <div>
      <table style={{width: '100%'}}>
        <thead>
          <tr>
            <th>#</th>
            <th>Name</th>
            <th>Email</th>
            <th>Phone</th>
            <th>Funct</th>
          </tr>
        </thead>
        <tbody>
        {contacts.data && contacts.data.length > 0 && contacts.data.map(n => {
          return (
            <tr key={n.id}>
              <td>{n.id}</td>
              <td>{n.name}</td>
              <td>{n.email}</td>
              <td>{n.phone}</td>
              <td>...buttons goes here</td>
            </tr>
          )

        })

        }
        </tbody>

      </table>

    </div>

  )

}

}


export default contacts;
