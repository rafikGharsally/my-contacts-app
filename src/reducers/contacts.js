import * as types from '../constants/types';

export default function contacts(state = {
  data: null,
  isFetching: false,
  error: false,
}, action) {
  switch (action.type) {
    case types.FETCH_CONTACTS: {
      return {
        ...state,
        data: null,
        error: false,
        isFetching: true
      };
    }
    case types.FETCH_CONTACTS_SUCCESS:
      return {
        ...state,
        data: action.payload,
        error: false,
        isFetching: false
      };
    default:
      return state;
  }
}