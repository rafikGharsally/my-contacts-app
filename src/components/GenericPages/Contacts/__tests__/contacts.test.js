import React from 'react';
import contacts  from '../contacts';
import renderer from "react-test-renderer";


describe('Contacts test suit', () => {

  let fakeContacts = {
    data:[
      {
        "id": 1,
        "name": "Rafik Gharsalli",
        "job_title": "software engineer",
        "address": "1234 test test",
        "picture": "",
        "email": "rafikgharsally@gmail.com",
        "phone": "5145891865"
      }
    ],
    error:false
  };

  // TODO fix tests
  // component need to be refactored to make it work ---- because the component is wrapped 'withStyles' o
  // or add create a theme provider

  // it('renders without crashing --- here', () => {
  //
  //   const wrapper = shallow(
  //     <contacts contacts={fakeContacts}/>
  //   );
  //   wrapper.find('#view').first().simulate('click');
  //   expect(history.push).toHaveBeenCalledWith('/view');
  // });


  it('renders and matches our snapshot', () => {
    const component = renderer.create((<contacts contacts={fakeContacts} />));
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });


});