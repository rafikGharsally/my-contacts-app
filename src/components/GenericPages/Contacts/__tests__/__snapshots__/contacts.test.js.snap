// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contacts test suit renders and matches our snapshot 1`] = `
<contacts
  contacts={
    Object {
      "data": Array [
        Object {
          "address": "1234 test test",
          "email": "rafikgharsally@gmail.com",
          "id": 1,
          "job_title": "software engineer",
          "name": "Rafik Gharsalli",
          "phone": "5145891865",
          "picture": "",
        },
      ],
      "error": false,
    }
  }
/>
`;
